### PR TITLE
Sf3 choice loader fix

### PR DESCRIPTION
**Pull request by Mangetsu**
Add ChoiceLoader class type for multi choice selects in SonataAdminBundle ModelsToArrayTransformer.

### CODE REVIEW

File: Form/DataTransformer/ModelsToArrayTransformer.php [hunk 1]
```diff
@@ -50,7 +51,7 @@ class ModelsToArrayTransformer implements DataTransformerInterface
      */
     public function __construct($choiceList, ModelManagerInterface $modelManager, $class)
     {
-        if (!$choiceList instanceof ModelChoiceList && !$choiceList instanceof LazyChoiceList) {
+        if (!$choiceList instanceof ModelChoiceList && !$choiceList instanceof ModelChoiceLoader && !$choiceList instanceof LazyChoiceList) {
             throw new RuntimeException('First param passed to ModelsToArrayTransformer should be instance of ModelChoiceList or LazyChoiceList');
```
Comment (greg0ire) [inline Form/DataTransformer/ModelsToArrayTransformer.php]: The exception message should be updated too.

File: Form/DataTransformer/ModelsToArrayTransformer.php [hunk 2]
```diff
@@ -50,7 +51,7 @@ class ModelsToArrayTransformer implements DataTransformerInterface
      */
     public function __construct($choiceList, ModelManagerInterface $modelManager, $class)
     {
-        if (!$choiceList instanceof ModelChoiceList && !$choiceList instanceof LazyChoiceList) {
+        if (!$choiceList instanceof ModelChoiceList && !$choiceList instanceof ModelChoiceLoader && !$choiceList instanceof LazyChoiceList) {
```
Comment (greg0ire) [inline Form/DataTransformer/ModelsToArrayTransformer.php]: Please linebreak this.

File: Form/Type/ModelType.php [hunk 3]
```diff
@@ -51,7 +51,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
     {
         if ($options['multiple']) {
             if (array_key_exists('choice_loader', $options) && $options['choice_loader'] !== null) { // SF2.7+
-                $builder->addViewTransformer(new ModelsToArrayTransformer($options['choice_list'], $options['model_manager'], $options['class']), true);
+                $builder->addViewTransformer(new ModelsToArrayTransformer($options['choice_loader'], $options['model_manager'], $options['class']), true);
```
Comment (greg0ire) [inline Form/Type/ModelType.php]: Please linebreak this.

**Comment by greg0ire** [comment]
Please use good commit messages : http://rakeroutes.com/blog/deliberate-git/ "update file.txt" is paraphrasing.

**Comment by greg0ire** [comment]
ping @Mangetsu

File: Form/DataTransformer/ModelsToArrayTransformer.php [hunk 4]
```diff
@@ -50,9 +51,8 @@ class ModelsToArrayTransformer implements DataTransformerInterface
      */
     public function __construct($choiceList, ModelManagerInterface $modelManager, $class)
     {
-        if (!$choiceList instanceof ModelChoiceList && !$choiceList instanceof LazyChoiceList) {
-            throw new RuntimeException('First param passed to ModelsToArrayTransformer should be instance of ModelChoiceList or LazyChoiceList');
-        }
+        if (!$choiceList instanceof ModelChoiceList && !$choiceList instanceof ModelChoiceLoader && !$choiceList instanceof LazyChoiceList) {
+            throw new RuntimeException('First param passed to ModelsToArrayTransformer should be instance of ModelChoiceLoader or ModelChoiceList or LazyChoiceList');        }
```
Comment (greg0ire) [inline Form/DataTransformer/ModelsToArrayTransformer.php]: Can you linebreak both lines? They are way too long.

**Comment by greg0ire** [comment]
Ok it's better. After you're done, run `make cs`, it should "make the build green again" #greg0ire2016 Then, run `git reset origin/master` to forget your commits. Then, use git commit to commit your changes into one commit, and please add a meaningful commit message ("update file.txt" is not going to get merged, you have to explain what you are doing, and why you are doing it).

> Please use good commit messages : http://rakeroutes.com/blog/deliberate-git/ "update file.txt" is paraphrasing.

This is really important, please take the time to read and apply for every commit you do. You will thank me later.

**Comment by Mangetsu** [comment]
thank you @greg0ire for the assistance, I made the changes and updated my commit message.
Everything is ok now ?

**Comment by greg0ire** [comment]
No it isn't : there should be 1 commit, not 6, and the commit message must be descriptive, not "Update ModelsToArrayTransformer.php". And the build should be green.

**Comment by greg0ire** [comment]
Let's go step by step. First, try to make the build green : 

```
make cs
git add --patch
git commit
git push
```

**Comment by greg0ire** [comment]
Ask for help if you need some.

**Pull request closed without merging**